artifact-ast-serialization/src/test/java/org/mule/test/integration/ast/serialization/OldArtifactAstWithNewSerializerAndRuntimeExtensionModelsTestCase.java

```diff
@@ -0,0 +1,148 @@
+import org.mule.runtime.ast.api.xml.AstXmlParser;
+import org.mule.runtime.ast.internal.serialization.json.JsonArtifactAstSerializerFormat;
+import org.mule.runtime.core.api.extension.provider.MuleExtensionModelProvider;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import io.qameta.allure.Feature;
+import io.qameta.allure.Issue;
+import io.qameta.allure.Story;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+@Feature(AST_SERIALIZATION)
+@Story(AST_SERIALIZATION_END_TO_END)
+public class OldArtifactAstWithNewSerializerAndRuntimeExtensionModelsTestCase {
+
+  @Parameters(name = "version: {0}")
+  public static Iterable<String> data() {
+    return new JsonArtifactAstSerializerFormat().getAvailableVersions();
+  }
+
+  private static final ExtensionModel RUNTIME_EXTENSION_MODEL = MuleExtensionModelProvider.getExtensionModel();
+  private static final String APP_RESOURCE_PATH = "apps/app-with-choice.xml";
+
+  @Parameter
+  public String version;
+
+  private final ArtifactAstSerializerProvider serializerProvider = new ArtifactAstSerializerProvider();
+  private InputStream serializedAst;
+
+  @Before
+  public void setup() {
+    ArtifactAst artifactAst = createArtifactFromXmlFile(APP_RESOURCE_PATH);
+    ArtifactAstSerializer serializer = serializerProvider.getSerializer("JSON", version);
+    serializedAst = serializer.serialize(artifactAst);
+  }
+
+  @Test
+  @Issue("W-14722981")
+  public void parametersAllowingExpressionsWithoutMarkersAreSerializedCorrectly()
+      throws IOException {
+    consumeHeaderLine(serializedAst);
+
+    JSONObject choiceParameterValueJson = getChoiceParameterValue(serializedAst);
+    if ("1.0".equals(version)) {
+      // When serializing with 1.0 the value must not be fixed
+      assertThat(choiceParameterValueJson.has("expression"), is(false));
+      assertThat(choiceParameterValueJson.has("aBoolean"), is(true));
+      assertThat(choiceParameterValueJson.getBoolean("aBoolean"), is(false));
+    } else {
+      assertThat(choiceParameterValueJson.has("expression"), is(true));
+      assertThat(choiceParameterValueJson.getString("expression"), is("0 != 1"));
+    }
+  }
+
+  @Test
+  @Issue("W-14722981")
+  public void parametersAllowingExpressionsWithoutMarkersAreDeserializedCorrectly()
+      throws IOException {
+    ArtifactAst deserializedAst = serializerProvider.getDeserializer().deserialize(serializedAst, this::resolveExtensionModel);
+    Either<String, Boolean> choiceParameterValue = getChoiceParameterValue(deserializedAst);
+
+    // No matter the serialized version, it must always be deserialized to the correct value
+    assertThat(choiceParameterValue.isLeft(), is(true));
```

Comment:
can you implement an matcher for either? If this assertion fails it will log something like `expected true but was false`, with a matcher it would be `expected an either with a left value XXX but was right value YYY`